be/src/formats/orc/apache-orc/c++/src/StripeStream.cc

```diff
@@ -100,11 +100,9 @@ std::unique_ptr<SeekableInputStream> StripeStreamsImpl::getStream(uint64_t colum
         const proto::Stream& stream = footer.streams(i);
         if (stream.has_kind() && stream.kind() == kind && stream.column() == static_cast<uint64_t>(columnId)) {
             uint64_t streamLength = stream.length();
-            uint64_t myBlock = shouldStream ? input.getNaturalReadSize() : streamLength;
-            // if we don't need that much data, why we read it.
-            if (streamLength < myBlock) {
-                myBlock = streamLength;
-            }
+            // uint64_t myBlock = shouldStream ? input.getNaturalReadSize() : streamLength;
+            // if we don't need that much data, why we read it?
+            uint64_t myBlock = std::min(input.getNaturalReadSize(), streamLength);
```

Comment:
```if shouldStream == false && input.getNaturalReadSize() < streamLength```
Current implementation will let be `myBlock` to be `input.getNaturalReadSize()`, but original implementation will be `streamLength`. Is this OK?

Reply:
Yes. it's ok. It's all about optimization but not correctness. I suspect this will trigger execessive read volume which I think it's quite unecessary.

Reply:
let's talk about this offline

Reply:
Yes, this is a bug. Previous version is correct. Some input stream can not be read in stream way like string dictionary. It 's has to be fully loaded.